evm/logic/call.py

```diff
@@ -348,3 +348,13 @@ def get_call_params(self, computation):
         if computation.msg.is_static and value != 0:
             raise WriteProtection("Cannot modify state while inside of a STATICCALL context")
         return call_params
+
+
+class CallSharding(CallByzantium):
+    def compute_msg_extra_gas(self, computation, gas, to, value):
+        with computation.state_db(read_only=True) as state_db:
+            account_is_empty = state_db.account_is_empty(to)
+
+        transfer_gas_fee = constants.GAS_CALLVALUE if value else 0
+        create_gas_fee = constants.GAS_NEWACCOUNT if not account_is_empty else 0
```

Comment:
It seems `create_gas_fee` should only be charged if the account is empty?

But with `FlatTrieBackend` IMHO we should probably review/modify this `create_gas_fee` as described here: https://github.com/ethereum/py-evm/issues/289#issuecomment-359379247